### PR TITLE
EPUI-4 create parking lot

### PR DESCRIPTION
**Pull request by Lewissp312**
# Pull Request

## Ticket link
https://trello.com/c/lOwPgXmC

## Why have you made these changes? What problem it fixes?
N/A

## What does this implement? Explain your changes
Enables the admin to register parking lots

## Is it better than when you found it?
N/A

## Comments
N/A

### CODE REVIEW

File: src/views/components/loginForm.js [hunk 1]
```diff
@@ -70,7 +69,7 @@ export default function LoginForm() {
           {/* Display error message if any */}
           {error && (
             <Alert data-test-id="login-alert" className='mt-4' variant="danger">
-              {"Failed to login: "+ error}
+              {error}
```
Comment (IgorSteps) [inline src/views/components/loginForm.js]: Bring that back pls

File: src/views/components/createParkingLotForm.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+          <Form.Group controlId="formBasicLotName">
+            <Form.Label>Name: </Form.Label>
+            <Form.Control 
+              type="text" 
+              placeholder="Enter name" 
+              value={name} 
+              onChange={(e) => setName(e.target.value)} 
+              required 
+              data-test-id="lot-name-input"
+            />
```
Comment (IgorSteps) [inline src/views/components/createParkingLotForm.js:34]: Let's add `required` tag  - it checks that the input field is not empty when user click submit

File: src/views/components/createParkingLotForm.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+            <Form.Label>Capacity: </Form.Label>
+            <Form.Control 
+              type="number" 
+              placeholder="Enter capacity" 
+              value={capacity} 
+              onChange={(e) => setCapacity(e.target.value)} 
+              min={30}
+              max={1000}
+              required 
+              data-test-id="lot-capacity-input"
```
Comment (IgorSteps) [inline src/views/components/createParkingLotForm.js:47]: Same here

File: src/models/createParkingLotService.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+            'Authorization': 'Bearer ' + token,
+            'Content-Type': 'application/json'
+        }
+    });
+    console.log(response);
+    console.debug("Got response data for creating a parking lot", response.data)
+    return response.data;
+  } catch (error) {
+    console.error('Failed to create parking lot', error.response);
+    throw new Error(error.response.data.error || 'An error occurred');
```
Comment (IgorSteps) [inline src/models/createParkingLotService.js]: Are you sure it's not `error.response.data`?

**Review by IgorSteps — CHANGES_REQUESTED** [review]
Great work, some comments

File: cypress/e2e/createParkingLot.cy.js [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+      // ---
+      cy.get('[data-test-id="lot-name-input"]').type('Test lot 1');
+      cy.get('[data-test-id="lot-capacity-input"]').type("50");
+      cy.get('[data-test-id="create-parking-lot-submit-button"]').click();
+  
+      // ------
+      // ASSERT
+      // ------
+
+      //cy.get('[data-test-id=create-park-request-success-alert]').should('contain', 'Successfully created parking request');
```
Comment (IgorSteps) [inline cypress/e2e/createParkingLot.cy.js]: Why is this commented out? Are we not checking if the create operation succeeded?

File: cypress/e2e/createParkingLot.cy.js [hunk 6]
```diff
@@ -0,0 +1,54 @@
+describe('Login Form', () => {
```
Comment (IgorSteps) [inline cypress/e2e/createParkingLot.cy.js]: Change the name please

**Review by IgorSteps — COMMENTED** [review]
Nice, some comments, also don't forget about the `required` things from previous review

File: src/views/components/createParkingLotForm.js (hunk 2 above)
Comment (Lewissp312) [reply src/views/components/createParkingLotForm.js:34]: I believe I have added the required tag there, or is there something else you'd like me to do with it?

Comment (IgorSteps) [reply src/views/components/createParkingLotForm.js:34]: I am blind, sorry

**Review by IgorSteps — APPROVED** [review]
⚡

**Pull request merged**